vm/package/deploy.rs

```diff
@@ -0,0 +1,141 @@
+    fn deserialize<D: Deserializer<'de>>(deserializer: D) -> Result<Self, D::Error> {
+        // Parse the request from a string into a value.
+        let request = serde_json::Value::deserialize(deserializer)?;
+        // Recover the leaf.
+        Ok(Self::new(
+            // Retrieve the program.
+            serde_json::from_value(request["deployment"].clone()).map_err(de::Error::custom)?,
+        ))
+    }
+}
+
+pub struct DeployResponse<N: Network> {
+    deployment: Deployment<N>,
+}
+
+impl<N: Network> DeployResponse<N> {
+    /// Initializes a new deploy response.
+    pub const fn new(deployment: Deployment<N>) -> Self {
+        Self { deployment }
+    }
+
+    /// Returns the program ID.
+    pub const fn deployment(&self) -> &Deployment<N> {
+        &self.deployment
+    }
+}
+
+impl<N: Network> Serialize for DeployResponse<N> {
+    /// Serializes the deploy response into string or bytes.
+    fn serialize<S: Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
+        let mut response = serializer.serialize_struct("DeployResponse", 1)?;
+        response.serialize_field("deployment", &self.deployment)?;
+        response.end()
+    }
+}
+
+impl<'de, N: Network> Deserialize<'de> for DeployResponse<N> {
+    /// Deserializes the deploy response from a string or bytes.
+    fn deserialize<D: Deserializer<'de>>(deserializer: D) -> Result<Self, D::Error> {
+        // Parse the response from a string into a value.
+        let response = serde_json::Value::deserialize(deserializer)?;
+        // Recover the leaf.
+        Ok(Self::new(
+            // Retrieve the program ID.
+            serde_json::from_value(response["program_id"].clone()).map_err(de::Error::custom)?,
+        ))
+    }
+}
+
+impl<N: Network> Package<N> {
+    pub fn deploy<A: crate::circuit::Aleo<Network = N, BaseField = N::Field>>(
+        &self,
+        endpoint: Option<String>,
+    ) -> Result<()> {
+        // Retrieve the main program.
+        let program = self.program();
+
+        // Retrieve the program ID.
+        let program_id = program.id();
+
+        // Construct the process.
+        let process = Process::<N>::load()?;
+        let rng = &mut test_crypto_rng();
+
+        // Make the deploy
+        let deployment = process.deploy::<A, _>(program, rng).unwrap();
+
+        match endpoint {
+            Some(ref endpoint) => {
+                // Prepare the request
+                let request = DeployRequest::new(deployment);
+                // Load the proving and verifying keys.
+                let response = request.send(endpoint)?;
+                // Ensure the program ID matches.
+                ensure!(
+                    response.deployment.program_id() == program_id,
+                    "Program ID mismatch: {} != {program_id}",
+                    response.deployment.program_id()
+                );
+            }
+            //FIXME correctly manage this case
+            None => {
+                todo!()
```

Comment:
@raychu86 we should see what to do in this case, from our side we're not sure if we should have a default value.